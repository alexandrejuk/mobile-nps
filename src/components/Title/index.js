import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

import styles from './style.module.css'

const Title = ({
  color,
  size,
  text,
  textAlign,
  type,
}) => (
  <h1
    className={classNames(
      styles[color],
      styles[size],
      styles[textAlign],
      styles.title,
      styles[type],
    )}
  >
    {text}
  </h1>
)

Title.propTypes = {
  textAlign: PropTypes.PropTypes.oneOf([
    'center',
    'left',
    'right',
  ]),
  color: PropTypes.PropTypes.oneOf([
    'primary',
    'secondary',
  ]),
  size: PropTypes.PropTypes.oneOf([
    'medium',
    'small',
    'large',
  ]),
  text: PropTypes.string.isRequired,
  type: PropTypes.oneOf([
    'bold',
    'normal',
  ]),
}

Title.defaultProps = {
  textAlign: 'left',
  color: 'primary',
  size: 'medium',
  type: 'normal',
}

export default Title
