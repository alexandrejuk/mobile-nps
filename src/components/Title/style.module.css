@import '../../assets/css/colors.css';

.title {
  font-family: 'Rubik', Arial, Helvetica, sans-serif;
}

.normal {
  font-size: 14px;
  font-weight: normal;
}


.medium {
  font-size: 16px;
  font-weight: normal;
}

.small {
  font-size: 12px;
  font-weight: normal;
}

.large {
  font-size: 24px;
  font-weight: normal;
}

.primary {
  color: var(--color-ship-gray);
}

.bold {
  font-weight: bold;
}

.center {
  text-align: center;
}

.left {
  text-align: left;
}

.right {
  text-align: right;
}

.secondary {
  color: var(--color-jumbo);
}

.cardinal {
  color: var(--color-cardinal)
}

.medium-1 {
  font-size: 20px;
}
