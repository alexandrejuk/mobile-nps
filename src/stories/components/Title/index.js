import React from 'react'
import { Title } from '../../../components'

export default {
  title: 'Components/Title',
}

const titles = [
  {
    size: 'small',
    text: 'Small title!',
  },
  {
    size: 'medium',
    text: 'Medium title!',
  },
  {
    size: 'large',
    text: 'Large title!',
  },
]

export const Default = () => titles.map( title => (
  <Title {...title} />
))

export const DefaultBold = () => titles.map( title => (
  <Title {...title} type="bold" />
))
