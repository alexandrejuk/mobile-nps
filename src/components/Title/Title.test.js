import React from 'react'
import { render } from '@testing-library/react'

import Title from './'

describe('Title component', () => {
  it('should render correctly', () => {
    const { container } = render(
      <Title
        text="I am a title text"
      />
    )
    expect(container).toMatchSnapshot()
  })

  it('should render correctly with small size', () => {
    const { container } = render(
      <Title
        text="I am a title text small"
        size="small"
      />
    )
    expect(container).toMatchSnapshot()
  })

  it('should render correctly with large size', () => {
    const { container } = render(
      <Title
        text="I am a title text large"
        size="large"
      />
    )
    expect(container).toMatchSnapshot()
  })

  it('should render correctly with small size and right text aligin', () => {
    const { container } = render(
      <Title
        text="I am a title text small right align"
        textAlign="right"
        size="small"
      />
    )
    expect(container).toMatchSnapshot()
  })

  it('should render correctly with large size and center text aligin', () => {
    const { container } = render(
      <Title
        text="I am a title text large and center"
        textAlign="center"
        size="large"
      />
    )
    expect(container).toMatchSnapshot()
  })

  it('should render correctly with bold type', () => {
    const { container } = render(
      <Title
        text="I am a title text large"
        type="bold"
      />
    )
    expect(container).toMatchSnapshot()
  })
})
