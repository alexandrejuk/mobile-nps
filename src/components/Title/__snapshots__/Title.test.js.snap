// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title component should render correctly 1`] = `
<div>
  <h1
    class="primary medium left title normal"
  >
    I am a title text
  </h1>
</div>
`;

exports[`Title component should render correctly with bold type 1`] = `
<div>
  <h1
    class="primary medium left title bold"
  >
    I am a title text large
  </h1>
</div>
`;

exports[`Title component should render correctly with large size 1`] = `
<div>
  <h1
    class="primary large left title normal"
  >
    I am a title text large
  </h1>
</div>
`;

exports[`Title component should render correctly with large size and center text aligin 1`] = `
<div>
  <h1
    class="primary large center title normal"
  >
    I am a title text large and center
  </h1>
</div>
`;

exports[`Title component should render correctly with small size 1`] = `
<div>
  <h1
    class="primary small left title normal"
  >
    I am a title text small
  </h1>
</div>
`;

exports[`Title component should render correctly with small size and right text aligin 1`] = `
<div>
  <h1
    class="primary small right title normal"
  >
    I am a title text small right align
  </h1>
</div>
`;
